### PR TITLE
Azure Search Fixes

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR makes the following changes.

- Adds `IndexDocumentsOptions` which is used as an option bag in indexing operations.
~~- Moved other options models into new options package.~~
  ~~- `AutocompleteOptions`, `SearchOptions`, and `SuggestOptions`.~~
- Removed implementation classes from public APIs in `SearchPagedResponse` and `SuggestPagedResponse`.
- Moved `FieldBuilder` and related classes into samples.

### CODE REVIEW

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/options/package-info.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * Package containing the classes which serve as options to APIs.
+ */
+package com.azure.search.documents.options;
```
Comment (sima-zhu) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/options/package-info.java:7]: We have several other options in indexes.
Do we only put seachClient related options in option bag?

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/util/SearchPagedResponse.java [hunk 2]
```diff
@@ -37,40 +31,24 @@
...
-    public SearchPagedResponse(SimpleResponse<SearchDocumentsResult> documentSearchResponse,
-        SearchServiceVersion serviceVersion) {
-        super(documentSearchResponse.getRequest(),
-            documentSearchResponse.getStatusCode(),
-            documentSearchResponse.getHeaders(),
-            documentSearchResponse.getValue().getResults().stream()
-                .map(SearchResultConverter::map).collect(Collectors.toList()),
-            createContinuationToken(documentSearchResponse, serviceVersion),
-            null);
+    public SearchPagedResponse(HttpRequest request, int statusCode, HttpHeaders headers,
```
Comment (sima-zhu) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/util/SearchPagedResponse.java:43]: Can we pass new PagedResponseBase here? It has too many parameters.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/options/package-info.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/options/package-info.java:7]: Moving classes back to original location as they don't fit the guideline for option bags.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/options/IndexDocumentsOptions.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+package com.azure.search.documents.options;
+
+import com.azure.core.annotation.Fluent;
+
+/**
+ * Options for document index operations.
+ */
+@Fluent
+public class IndexDocumentsOptions {
+    private boolean throwOnAnyError = false;
```
Comment (srnagar) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/options/IndexDocumentsOptions.java]: Should the default be `true`?

Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/options/IndexDocumentsOptions.java]: .NET is using false by default. https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/search/Azure.Search.Documents/src/Options/IndexDocumentsOptions.cs#L19

@tg-msft @heaths could explain the reasoning better.

Comment (sima-zhu) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/options/IndexDocumentsOptions.java]: Personally prefer to take `true`, as we have retry helper inside of IndexBatchException.

**Pull request merged**